secrets/gcpssm/gcp_ssm.go

```diff
@@ -0,0 +1,168 @@
+package gcpssm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/0xPolygon/polygon-edge/secrets"
+	"github.com/hashicorp/go-hclog"
+	"os"
+
+	sm "cloud.google.com/go/secretmanager/apiv1"
+	smpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+)
+
+type GcpSsmManager struct {
+	// project id in which to store the secrets
+	projectID string
+	// gcp secrets manager client
+	client *sm.Client
+	// credential file path
+	credFilePath string
+	// logger instance
+	logger hclog.Logger
+	// context used in API calls
+	context context.Context
+	// node name is used to create unique secret id
+	nodeName string
+}
+
+func SecretsManagerFactory(
+	config *secrets.SecretsManagerConfig,
+	params *secrets.SecretsManagerParams,
+) (secrets.SecretsManager, error) {
+	// Check if project id is defined
+	if _, ok := config.Extra["project-id"]; !ok {
+		return nil, errors.New("no project-id variable specified")
+	}
+	// Check if gcp-ssm-cred is defined
+	if _, ok := config.Extra["gcp-ssm-cred"]; !ok {
+		return nil, errors.New("no gcp-ssm-cred variable specified")
+	}
+	// Check if the variables are present
+	if config.Name == "" || config.Extra["project-id"] == "" || config.Extra["gcp-ssm-cred"] == "" {
+		return nil, errors.New("name, project-id or gcp-ssm-cred can't be empty string")
```

Comment:
Typo here, `can't be an empty string`

Reply:
Fixed in b95eb9d2fcd7a213bc19ef0ab427cde103615642